static/app/components/replays/canvasReplayerPlugin.tsx

```diff
@@ -63,44 +63,35 @@ export function CanvasReplayerPlugin(events: eventWithTime[]) {
   function cloneCanvas(id: number, node: HTMLCanvasElement) {
     const cloneNode = node.cloneNode() as HTMLCanvasElement;
     canvases.set(id, cloneNode);
-    const el = document.createElement('img');
-    node.appendChild(el);
     document.adoptNode(cloneNode);
-    containers.set(id, el);
     return cloneNode;
   }
 
+  const promises: Promise<any>[] = [];
+  for (const event of events) {
+    if (
+      event.type === EventType.IncrementalSnapshot &&
+      event.data.source === IncrementalSource.CanvasMutation
+    ) {
+      promises.push(deserializeAndPreloadCanvasEvents(event.data, event));
+    }
+  }
+
   return {
     /**
      * When document is first built, we want to preload canvas events. After a
      * `canvas` element is built (in rrweb), insert an image element which will
      * be used to mirror the drawn canvas.
      */
-    onBuild: async (node, {id}) => {
+    onBuild: (node, {id}) => {
       if (!node) {
         return;
       }
 
-      if (node.nodeType === 9) {
-        // on full snapshot... deserialize all canvas events
-        const promises: Promise<any>[] = [];
-        containers.clear();
-        for (const event of events) {
-          if (
-            event.type === EventType.IncrementalSnapshot &&
-            event.data.source === IncrementalSource.CanvasMutation
-          ) {
-            promises.push(deserializeAndPreloadCanvasEvents(event.data, event));
-          }
-        }
-
-        await Promise.all(promises);
-      }
-
-      if (node.nodeName === 'CANVAS' && node instanceof HTMLCanvasElement) {
```

Comment:
I think `node instanceof HTMLCanvasElement` was returning false because node doesn't belong to current document. (This check here was more to appease TS anyway).